src/coprocessor/codec/mysql/time/mod.rs

```diff
@@ -89,13 +89,18 @@ fn ymd_hms_nanos<T: TimeZone>(
     secs: u32,
     nanos: u32,
 ) -> Result<DateTime<T>> {
-    tz.ymd_opt(year, month, day)
-        .and_hms_opt(hour, min, secs)
-        .single()
+    use chrono::NaiveDate;
+
+    // Note: We are not using `tz::from_ymd_opt` as suggested in chrono's README due to
+    // chronotope/chrono-tz #23.
+    // As a workaround, we first build a NaiveDate, then attach time zone information to it.
+    NaiveDate::from_ymd_opt(year, month, day)
+        .and_then(|date| date.and_hms_opt(hour, min, secs))
         .and_then(|t| t.checked_add_signed(Duration::nanoseconds(i64::from(nanos))))
+        .and_then(|datetime| tz.from_local_datetime(&datetime).earliest())
```

Comment:
Why is earliest chosen?

Reply:
Because MySQL choses earliest when there is ambiguous time.

For example, in America/Los_Angeles time zone: [After Sunday, 06 November, 2011 01:59:59 AM: Clocks were moved backward to become Sunday, 06 November, 2011 01:00:00 AM](https://timezonedb.com/time-zones/America/Los_Angeles). When giving `2011-11-06 01:59:59`, it can be interpreted to the following two:

- The time before transition: 2011-11-06 01:59:59 -7 (timestamp: 1320569999)

- The time after transition: 2011-11-06 01:59:59 -8 (timestamp: 1320573599)

In MySQL:

```
mysql> SET time_zone = 'America/Los_Angeles';
Query OK, 0 rows affected (0.03 sec)

mysql> select unix_timestamp('2011-11-06 01:59:59');
+---------------------------------------+
| unix_timestamp('2011-11-06 01:59:59') |
+---------------------------------------+
|                            1320569999 |
+---------------------------------------+
1 row in set (0.01 sec)

mysql> select unix_timestamp('2011-11-06 02:00:00');
+---------------------------------------+
| unix_timestamp('2011-11-06 02:00:00') |
+---------------------------------------+
|                            1320573600 |
+---------------------------------------+
1 row in set (0.00 sec)
```

Reply:
So `earliest` is equal to `single` when the conversion is unique?

Reply:
[According to its source code](https://github.com/chronotope/chrono/blob/94b43fa2e8bd43e7f42bb5b67afd1c3415b27683/src/offset/mod.rs#L49), yes.